py/torch_tensorrt/dynamo/runtime/_MutableTorchTensorRTModule.py

```diff
@@ -306,14 +340,26 @@ def compile(self) -> None:
         torch.cuda.empty_cache()
 
     def _validate_inputs(self, *args: Any, **kwargs: Any) -> None:
-        if (
-            not self.arg_inputs
-            or not MutableTorchTensorRTModule.check_inputs_equal(self.arg_inputs, args)
-            or not MutableTorchTensorRTModule.check_inputs_equal(
-                self.kwarg_inputs, kwargs
-            )
-        ):
+        try:
+            if (
+                not self.arg_inputs
+                or not MutableTorchTensorRTModule.check_inputs_equal(
+                    self.arg_inputs, args, dynamic_shapes=self.arg_dynamic_shapes
+                )
+                or not MutableTorchTensorRTModule.check_inputs_equal(
+                    self.kwarg_inputs, kwargs, dynamic_shapes=self.kwarg_dynamic_shapes
+                )
+            ):
+                logger.info("Input change detected.")
+                self.refit_state.set_state(RefitFlag.NEEDS_RECOMPILE)
+                self.store_inputs(args, kwargs)
+        except DynamicShapeOutOfRangeException as e:
             logger.info("Input change detected.")
+            logger.warning(e)
+            logger.warning("Recompiling the engine with static shape")
```

Comment:
Make this warning more verbose. Something like `Provided inputs are outside the set expected shape range, recompiling module for the provided input shapes (static)`